lib/tasks/config.rake

```diff
@@ -91,30 +91,34 @@ task :config => :environment do
 
     # just a single key was passed: print the value
     def run_single_key
-      setting = Setting.find_by_name(@key)
+      setting = Foreman.settings.find(@key)
       puts format_value(setting.settings_type, setting.value)
     end
 
     # key-value pairs were provided: try to change that
     def run_key_values
       @keys.each do |key|
         value = @key_values[key]
-        setting = Setting.find_by_name(key)
+        setting = Foreman.settings.find(key)
         if value == :unset
-          setting.value = nil
+          val = nil
         elsif complex_type?(setting.settings_type)
-          setting.value = typecast_value(setting.settings_type, value)
+          val = typecast_value(setting.settings_type, value)
         else
-          parse_and_set_string(setting, value)
+          val = value
```

Comment:
Is it okay that we don't use `val` anywhere?.. Or am I missing something.

Reply:
Fixed